enhancements/microshift/multus-cni-for-microshift.md

```diff
@@ -0,0 +1,401 @@
+It is primarily used with SR-IOV and as such out of scope for this enhancement mentioned only for completeness.
+
+
+### Implementation Details/Notes/Constraints [optional]
+
+First, it must be noted that Multus CNI itself is a meta-CNI. From high level perspective, its
+purpose is to call other CNIs according to the CNI configs supplied by the user in form of
+NetworkAttachmentDefinitions and Pod annotations.
+Any specific actions are related to the delegate CNIs itself. For example: creating veth pair,
+attaching one end of veth to the bridge and making the other end available within the Pod is
+responsibility of `bridge` CNI is this example.
+MicroShift team will create tests for CNIs that will be declared as supported. However these tests
+will not explore the breadth and depth of possible network setups, so ultimately responsible for
+correctness of the configuration is the user.
+
+#### Manifests
+
+MicroShift will provide Multus based on thin architecture because of the resource consumption
+(see alternatives for more information).
+Because of the differences with OpenShift (which uses thick Multus plugin), existing OpenShift
+manifests will require changes to make them suitable for MicroShift. These manifests will reside
+in MicroShift repository.
+
+Updating necessary image references will be part of existing rebase procedure.
+
+#### RPM package
+
+RPM spec to build `microshift-networking-multus` RPM will be part of existing `microshift.spec` file.
+RPM will include:
+- manifests required to deploy Multus on MicroShift
+- CRI-O drop-in config to use Multus instead of ovn-kubernetes
+- greenboot healthcheck script
+- cleanup script plugin
+
+#### Container images: Multus and network plugins
+
+Because Multus image used by OpenShift has size that is not acceptable for edge devices, new image
+will be prepared and will only include relevant artifacts such as entrypoint script/binary and
+Multus CNI binary (which is copied to host's `/opt/cni/bin`).
+
+To supply network plugins (CNIs) such as bridge, vlan, macvlan, and so on a new image will be prepared
+so MicroShift uses the same binaries as OpenShift (alternative is using RHEL's networkplugins RPM - see alternatives).
+This image will also contain IPAM binaries such as `static`, `dynamic (DHCP)`, and `host-local`.
+
+#### Hypershift [optional]
+
+No, enhancement is MicroShift specific.
+
+### Risks and Mitigations
+
+Taking `bridge` CNI as an example: when the bridge interface does not exist, it will be created
+when a Pod requiring that interface is created.
+If user expects something else will create the interface, they will need to configure system to start
+MicroShift after other services.
+
+Multus using thin plugin architecture creates a kubeconfig when the DaemonSet starts and copies it
+out of Pod to location on disk so the Multus CNI binary can use it to get NetworkAttachmentDefinitions
```

Comment:
It's `/opt/cni/bin`, I'd say it's **the** location. It can be configured with crio's `plugin_dirs` but it's the directory where ovn-k's binary already exists so I would not change it.

Reply:
OK, let's specify the path in the doc.

```suggestion
out of Pod to `/opt/cni/bin` on the host filesystem so the Multus CNI binary can use it to get NetworkAttachmentDefinitions
```